test/core/tsi/s2a/s2a_test_util.cc

```diff
@@ -105,6 +106,36 @@ std::vector<uint8_t> chacha_poly_empty_record_bytes = {
     0x17, 0x03, 0x03, 0x00, 0x11, 0xef, 0x8f, 0x7a, 0x42, 0x8d, 0xdc,
     0x84, 0xee, 0x59, 0x68, 0xcd, 0x63, 0x06, 0xbf, 0x1d, 0x2d, 0x1b};
 
+void verify_half_connections(uint16_t ciphersuite, s2a_crypter* crypter,
+                             std::vector<uint8_t>& expected_traffic_secret) {
+  GPR_ASSERT(crypter != nullptr);
+  uint8_t* expected_nonce = nullptr;
+  switch (ciphersuite) {
+    case kTlsAes128GcmSha256:
+      expected_nonce = aes_128_gcm_nonce_bytes.data();
+      break;
+    case kTlsAes256GcmSha384:
+      expected_nonce = aes_256_gcm_nonce_bytes.data();
+      break;
+    case kTlsChacha20Poly1305Sha256:
+      expected_nonce = chacha_poly_nonce_bytes.data();
+      break;
+    default:
+      gpr_log(GPR_ERROR, kS2AUnsupportedCiphersuite);
+      abort();
+  }
+  check_half_connection(
+      crypter, /** in_half_connection **/ true,
+      /** expected_sequence **/ 0, expected_traffic_secret.size(),
+      expected_traffic_secret.data(),
+      /** expected nonce size **/ 12, expected_nonce, SSL3_RT_HEADER_LENGTH);
```

Comment:
Do we have a const for `12`? It can be local to this file or function.

Reply:
In the next commit, I've deduced the expected nonce size from the test vector (rather than using `12` directly).